sklearn/tree/tests/test_tree.py

```diff
@@ -2658,7 +2658,10 @@ def test_regression_tree_missing_values_toy(X, criterion):
     y = np.arange(6)
 
     tree = DecisionTreeRegressor(criterion=criterion, random_state=0).fit(X, y)
-    assert all(tree.tree_.impurity >= 0)  # MSE should always be positive
+    tree_ref = clone(tree).fit(y.reshape(-1, 1), y)
+    # assert all(tree.tree_.impurity >= 0)  # MSE should always be positive
```

Comment:
@lesteve are you sure that this was not failing on `main`. Indeed, this is the example that I and @adam2392 used to be sure that the impurity of the tree make sense. The error in the statistics implied a negative MSE that is impossible.

Reply:
Sorry I did not mean to comment this test ...